### PR TITLE
webui: explain-strength-requirement

### PR DESCRIPTION
**Pull request by or2e**
- when trying to equip

image

- shop

image

- inventory

image

- automatic unequip items with a requirement, when `respec, retire or setTournament`

### CODE REVIEW

File: src/WebUI/src/services/characters-service.spec.ts [hunk 1]
```diff
@@ -146,7 +150,7 @@ it('updateCharacterCharacteristics', async () => {
 it.each([
   [0, 0],
   [1, 0],
-  [2, 3538],
+  [2, 152442],
```
Comment (or2e) [inline src/WebUI/src/services/characters-service.spec.ts:153]: changed after an [edit ](https://github.com/verdie-g/crpg/commit/61aa77967ff57ede4d6c09597d65ec0e8efd271d)

btw, we can store shared test-cases for these things in json(json5) format

File: src/WebUI/src/composables/character/use-inventory-dnd.ts [hunk 2]
```diff
@@ -1,26 +1,41 @@
...
+
+    // TODO: to service
     if (item.rank < 0) {
       notify(t('character.inventory.item.broken.notify.warning'), NotificationType.Warning);
       return;
     }
 
+    if (validateItemNotMeetRequirement(item.baseItem, characterCharacteristics.value)) {
+      notify(t('character.inventory.item.requirement.notify.warning'), NotificationType.Warning);
+      return;
```
[the remark this answers is not in the document]
Comment (verdie-g) [reply src/WebUI/src/composables/character/use-inventory-dnd.ts:36]: https://github.com/verdie-g/crpg/blob/master/src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs#L448

File: src/WebUI/src/services/characters-service.spec.ts [hunk 3]
```diff
@@ -146,7 +150,7 @@ it('updateCharacterCharacteristics', async () => {
 it.each([
   [0, 0],
   [1, 0],
-  [2, 3538],
+  [2, 388],
```
Comment (or2e) [inline src/WebUI/src/services/characters-service.spec.ts:153]: changed after an [edit](https://github.com/verdie-g/crpg/commit/61aa77967ff57ede4d6c09597d65ec0e8efd271d)

btw, we can store shared test-cases for these things in json(json5) format

File: src/WebUI/src/composables/character/use-inventory-dnd.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (or2e) [reply src/WebUI/src/composables/character/use-inventory-dnd.ts:36]: I figured out what to do, an icon similar to a broken item

**Comment by or2e** [comment]
image

**Review by verdie-g — APPROVED** [review]

**Pull request merged**